### PR TITLE
added date picker widget for forms

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

File: frontend/src/MuiForms/DateSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      onChange(dateInput.toDate().toISOString().split('T')[0])
+    }
+    if (dateInput === null) {
+      onChange('')
+    }
+  }
+
+  const displayValue = (dateValue: string) => {
+    const [year, month, date] = dateValue.split('-')
+    return `${date}/${month}/${year}`
```
[the remark this answers is not in the document]
Comment (a3957273) [reply frontend/src/MuiForms/DateSelector.tsx:37]: I also don't think we should be changing the display value of this.  If we're storing it as `DD-MM-YYYY` then we should display it as `DD-MM-YYYY`.

[the remark this answers is not in the document]
Comment (a3957273) [reply frontend/src/MuiForms/DateSelector.tsx:37]: Also, just in general, you don't want to format dates yourself but use the internalisation libraries within JavaScript.  Different countries will format dates very differently.

File: frontend/src/MuiForms/DateSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  InputProps?: any
+}
+
+export default function DateInput(props: CustomTextInputProps) {
+  const { onChange, value, label, formContext } = props
+
+  const theme = useTheme()
+  const handleChange = (dateInput: Dayjs | null) => {
+    if (dateInput && dateInput.isValid()) {
+      onChange(dateInput.toDate().toISOString().split('T')[0])
```
Comment (a3957273) [inline frontend/src/MuiForms/DateSelector.tsx:28]: Using `.format()` is likely clearer than this.  E.g. `dateInput.format('YYYY-MM-DD')`.

File: frontend/src/MuiForms/DateSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  }
+
+  const displayValue = (dateValue: string) => {
+    const [year, month, date] = dateValue.split('-')
+    return `${date}/${month}/${year}`
+  }
+
+  return (
+    <Fragment key={label}>
```
Comment (a3957273) [inline frontend/src/MuiForms/DateSelector.tsx:41]: You don't use required anywhere here, so there's no way to tell if it's mandatory or not.

File: frontend/src/MuiForms/DateSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      <Typography fontWeight='bold'>{label}</Typography>
+      {formContext.editMode && (
+        <LocalizationProvider dateAdapter={AdapterDayjs} adapterLocale='en-gb'>
```
Comment (a3957273) [inline frontend/src/MuiForms/DateSelector.tsx:44]: This can probably be far higher in the React tree, e.g. up at the top in `_app.tsx`.

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: frontend/src/MuiForms/DateSelector.tsx (hunk 4 above)
Comment (ARADDCC002) [reply frontend/src/MuiForms/DateSelector.tsx:44]: I thought the documentation meant you had to had to individually wrap the date pickers with these providers but if we can do it at the top level that'd be even better.

Comment (a3957273) [reply frontend/src/MuiForms/DateSelector.tsx:44]: On a different project we managed to just put it at the top level, so can probably do that here too unless they've changed something.

**Review by a3957273 — APPROVED** [review]

**Pull request merged**